docs/investigations/rest-adapters/rest-adapter-for-graphql.md

```diff
@@ -0,0 +1,103 @@
+# Provide REST Adapter for Compass Graphql API
+Compass GraphQL API is not compatible with Application Connector REST API.
+To avoid rewriting integration layer for  Application Connector clients, we can provide adapter that translates
+Application Connector API to Compass API.
+
+
+# Reproducing tutorial steps
+This section describes required modifications, that needs to be done to ensure that Application Connector tutorial 
+described [here](https://kyma-project.io/docs/1.8/components/application-connector/#tutorials-tutorials)
+still works.
+
+1. Create a new Application
+
+In the old approach, Application CR was created manually in the Runtime. In the new approach, Agents read information from Compass 
+Director and creates appropriate Application CRs.
+There is an open question, what should happen for manually created Application CRs. Right now, Agent ignores them completely.
+If there is a requirement, that manually created Application should be automatically registered in the Compass, 
+a new Controller has to be implemented.
+If there is a requirement, that manually created Application CR should be handled by old services, this complicates implementation, 
+because status of TokenRequest will be populated differently, depending if Application CR comes from Agent or was created manually.
+
+AFAIK, creating application was a manual step usually performed in UI. Because of that, we should consider
+not introducing previously mentioned controller, and just instruct Admins to create application using Compass UI. 
+
+2. Get the client certificate
+
+When creating Token Request, a  Controller updates the status and provide token and configuration URL fields.
```

Comment:
I'm not sure if this component is still valid with the Compass scenarios and approach. The required changes actually depend on whether we support the creation of the applications on the runtimes.

If we go with the limitation that all applications must be registered via the compass then this controller needs to call the Compass API to obtain the token and it may be worth to just update it to support just the new GraphQL API. In this case we need to consider how this component will communicate with the Compass instance - maybe it should be replaced by the Runtime Agent (?) which already has a trusted relation established.